core/lib/multivm/src/versions/vm_latest/tracers/evm_deploy_tracer.rs

```
@@ -0,0 +1,106 @@
+        _bootloader_state: &mut BootloaderState,
+    ) -> TracerExecutionStatus {
+        // We check if ContractDeployer was called with provided evm bytecode.
+        // It is assumed that by that time the user has already paid for its size.
+        // So even if we do not revert the addition of the this bytecode it is not a ddos vector, since
+        // the payment is the same as if the bytecode publication was reverted.
+        let current_callstack = &state.local_state.callstack.current;
+
+        // Here we assume that the only case when PC is 0 at the start of the execution of the contract.
+        let known_code_storage_call = current_callstack.this_address == KNOWN_CODES_STORAGE_ADDRESS
+            && current_callstack.pc == 0
+            && current_callstack.msg_sender == CONTRACT_DEPLOYER_ADDRESS;
+
+        if !known_code_storage_call {
+            // Just continue executing
+            return TracerExecutionStatus::Continue;
+        }
+
+        // Now, we need to check whether it is indeed a call to publish EVM code.
+        let calldata_ptr =
+            state.local_state.registers[CALL_IMPLICIT_CALLDATA_FAT_PTR_REGISTER as usize];
+
+        let data = read_pointer(&state.memory, FatPointer::from_u256(calldata_ptr.value));
+
+        let contract = known_codes_contract();
+
+        if data.len() < 4 {
+            // Not interested
+            return TracerExecutionStatus::Continue;
+        }
+
+        let (signature, data) = data.split_at(4);
+
+        if signature
+            != contract
+                .function("publishEVMBytecode")
+                .unwrap()
+                .short_signature()
+        {
+            // Not interested
+            return TracerExecutionStatus::Continue;
+        }
+
+        let Ok(call_params) = contract
+            .function("publishEVMBytecode")
+            .unwrap()
+            .decode_input(data)
+        else {
+            // Not interested
```

Why is ignoring an error OK here?